devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -203,8 +215,18 @@ public boolean isSatisfiedBy(Task t) {
                             });
                 });
 
+        final AnalyticsExtension analyticsData = project.getExtensions().getByType(AnalyticsExtension.class);
+        final TaskProvider<QuarkusAnalyticsProd> quarkusAnalyticsProd = tasks.register(
+                QUARKUS_ANALYTICS_PROD_TASK_NAME, QuarkusAnalyticsProd.class);
+
         TaskProvider<QuarkusBuild> quarkusBuild = tasks.register(QUARKUS_BUILD_TASK_NAME, QuarkusBuild.class, build -> {
             build.dependsOn(quarkusBuildDependencies, quarkusBuildCacheableAppParts);
+            build.doLast(task -> {
+                quarkusAnalyticsProd.configure(analyticsDataTask -> {
```

Comment:
A build in Gradle strictly separates "task configuration" from "task execution". Mixing configuration with execution (here: doLast is execution, calling `configure`) likely causes problems. However, I suspect the added tasks can go away, when analytics data is sent directly from `BuildWorker`.

Reply:
All is done in the `BuildWorker` and the `QuarkusDev` task

Reply:
The code is in now. What do you think @snazy ?